src/server/services/wmts/qgswmts.cpp

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************
+                              qgswmts.cpp
+                              -------------------------
+  begin                : July 23 , 2018
+  copyright            : (C) 2018 by René-Luc D'Hont
+  email                : rldhont at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmodule.h"
+#include "qgswmtsutils.h"
+#include "qgswmtsgetcapabilities.h"
+#include "qgswmtsgettile.h"
+#include "qgswmtsgetfeatureinfo.h"
+
+#define QSTR_COMPARE( str, lit )\
+  (str.compare( QStringLiteral( lit ), Qt::CaseInsensitive ) == 0)
+
+namespace QgsWmts
+{
+
+  /**
+   * \ingroup server
+   * \class QgsWmts::Service
+   * \brief OGC web service specialized for WMTS
+   * \since QGIS 3.0
+   */
+  class Service: public QgsService
+  {
+    public:
+
+      /**
+       * Constructor for WMTS service.
+       * \param serverIface Interface for plugins.
+       */
+      Service( QgsServerInterface *serverIface )
+        : mServerIface( serverIface )
+      {}
+
+      QString name()    const override { return QStringLiteral( "WMTS" ); }
+      QString version() const override { return implementationVersion(); }
+
+      bool allowMethod( QgsServerRequest::Method method ) const override
+      {
+        return method == QgsServerRequest::GetMethod || method == QgsServerRequest::PostMethod;
+      }
+
+      void executeRequest( const QgsServerRequest &request, QgsServerResponse &response,
+                           const QgsProject *project ) override
+      {
+        Q_UNUSED( project );
+
+        QgsServerRequest::Parameters params = request.parameters();
+        QString versionString = params.value( QStringLiteral( "VERSION" ) );
```

Comment:
Can you use the `QgsServerParameters` class? This way, enumerators are used instead of string.